lib/phlex/proxy.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Phlex::Proxy < BasicObject
```

Comment:
We need to come up with a better name for this.

Reply:
Maybe just `Phlex::Inline`? 

Or `Phlex::InlineRender`, `Phlex::InlineProxy`, `Phlex::InlineRenderProxy`?